rtgui/editcallbacks.h

```diff
@@ -0,0 +1,256 @@
+    @param modifierKey Gtk's event modifier key (GDK_CONTROL_MASK | GDK_SHIFT_MASK | ...)
+    @return true if the preview has to be redrawn, false otherwise */
+    virtual bool drag2 (const int modifierKey);
+
+    /** @brief Triggered when the user is moving while holding down mouse button 3
+    @param modifierKey Gtk's event modifier key (GDK_CONTROL_MASK | GDK_SHIFT_MASK | ...)
+    @return true if the preview has to be redrawn, false otherwise */
+    virtual bool drag3 (const int modifierKey);
+
+    /** @brief Triggered when the user is releasing mouse button 1 while in action==ES_ACTION_PICKING mode
+    No modifier key is provided, since having a different modifier key than on button press will set picked to false.
+    @param picked True if the cursor is still above the the same object than on button pressed and with the same modifier keys.
+                  If false, the user moved the cursor away or the modifier key is different, so the element is considered as NOT selected.
+    @return true if the preview has to be redrawn, false otherwise */
+    virtual bool pick1 (const bool picked);
+
+    /** @brief Triggered when the user is releasing mouse button 2 while in action==ES_ACTION_PICKING mode
+    @param picked True if the cursor is still above the the same object than on button pressed and with the same modifier keys.
+                  If false, the user moved the cursor away or the modifier key is different, so the element is considered as NOT selected.
+    @return true if the preview has to be redrawn, false otherwise */
+    virtual bool pick2 (const bool picked);
+
+    /** @brief Triggered when the user is releasing mouse button 3 while in action==ES_ACTION_PICKING mode
+    @param picked True if the cursor is still above the the same object than on button pressed and with the same modifier keys.
+                  If false, the user moved the cursor away or the modifier key is different, so the element is considered as NOT selected.
+    @return true if the preview has to be redrawn, false otherwise */
+    virtual bool pick3 (const bool picked);
+
+    /** @brief Get the geometry to be shown to the user */
+    const std::vector<Geometry*>& getVisibleGeometry ();
+
+    /** @brief Get the geometry to be drawn in the "mouse over" channel, hidden from the user */
+    const std::vector<Geometry*>& getMouseOverGeometry ();
+};
+
+/** @brief Class to handle the furniture of data to the subscribers.
+ *
+ * It is admitted that only one Subscriber can ask data at a time. If the Subscriber is of type ET_PIPETTE, it will have to
+ * trigger the usual event so that the image will be reprocessed to compute the buffer of the current subscriber.
+ */
+class EditDataProvider
+{
+
+private:
+    EditSubscriber *currSubscriber;
+
+public:
+    int object;            /// ET_OBJECTS mode: Object detected under the cursor, 0 otherwise; ET_PIPETTE mode: 1 if above the image, 0 otherwise

```

Comment:
Public data breaks encapsulation. You could put it into an inner `struct` and add getters/setters.